### PR TITLE
SPORTS-189 Login page basic layout

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
* Add login form
* Integrate `login` API
* Store the tokens to `localStorage`

Screenshot from 2019-08-12 14-52-38

### CODE REVIEW

File: admin-app/src/utils/httpHelper.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    url: baseUrl,
+    method: 'POST',
+    data: {
+      query
+    }
+  });
+
+  const data = response && response.data && response.data.data && response.data.data[queryApi];
+
+  if (data && data.code && data.code === 200) {
```
Comment (SAGARACH65) [inline admin-app/src/utils/httpHelper.ts]: Can we use HTTPStatus codes instead of hard coding the values?

Comment (sanjeevkpandit) [reply admin-app/src/utils/httpHelper.ts]: Done.

File: admin-app/src/router.ts [hunk 2]
```diff
@@ -62,3 +65,29 @@ export default new Router({
     }
   ]
 });
+
+router.beforeEach((to, from, next) => {
+  if (to.matched.some((record) => record.meta.requiresAuth)) {
```
Comment (SAGARACH65) [inline admin-app/src/router.ts]:
```suggestion
  if (to.matched.some(record => record.meta.requiresAuth)) {
```

File: admin-app/src/router.ts [hunk 3]
```diff
@@ -62,3 +65,29 @@ export default new Router({
...
+    const rawUserSession = localStorage.getItem('userSession');
+
+    if (rawUserSession === null || !rawUserSession.length) {
```
Comment (SAGARACH65) [inline admin-app/src/router.ts]:
```suggestion
    if (!rawUserSession  || !rawUserSession.length) {
```

File: admin-app/src/services/auth.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Check login by username and password.
+ *
+ * @export
+ * @param {string} userEmail
+ * @param {string} userPassword
+ * @returns
+ */
+export async function checkLogin(userEmail: string, userPassword: string) {
```
Comment (SAGARACH65) [inline admin-app/src/services/auth.ts]: `checkIfLoggedIn` might be a better name

File: admin-app/src/services/auth.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  const queryApi = `login`;
```
Comment (SAGARACH65) [inline admin-app/src/services/auth.ts]:
```suggestion
  const queryAPI = `login`;
```

File: admin-app/src/services/storage.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+const SESSION_KEY = 'userSession';
+
+/**
+ * Set user session to the storage.
+ *
+ * @export
+ * @param {object} userSession
+ * @returns {boolean}
+ */
+export function setUserSession(userSession: object): boolean {
```
Comment (SAGARACH65) [inline admin-app/src/services/storage.ts]: Do we need to return a boolean here? The place where this function is getting called is               `storageService.setUserSession(params);`

File: admin-app/src/utils/httpHelper.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    url: baseUrl,
+    method: 'POST',
+    data: {
+      query
+    }
+  });
+
+  const data = response && response.data && response.data.data && response.data.data[queryApi];
+
+  if (data && data.code && data.code === httpStatusCodes.OK) {
```
Comment (SAGARACH65) [inline admin-app/src/utils/httpHelper.ts]:
```suggestion
  if (data && data.code === httpStatusCodes.OK) {
```

File: admin-app/src/views/Login.vue [hunk 8]
```diff
@@ -1,14 +1,130 @@
...
+  private logo: string = logoInvertedImage;
+
+  private beforeCreate() {
+    this.form = this.$form.createForm(this);
+  }
+
+  private handleSubmit(e: any) {
+    e.preventDefault();
+
+    this.form.validateFields((err: any, values: any) => {
```
Comment (SAGARACH65) [inline admin-app/src/views/Login.vue]: Can't we define an interface for values?

File: admin-app/src/utils/httpHelper.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    return data;
```
Comment (SAGARACH65) [inline admin-app/src/utils/httpHelper.ts]: The function description and the docs for this do nat have any types listed

File: admin-app/src/utils/httpHelper.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  });
+
+  const data = response && response.data && response.data.data && response.data.data[queryAPI];
+
+  if (data && data.code && data.code === httpStatusCodes.OK) {
+    return data;
+  }
+
+  throw new Error(data.message || 'An error occurred');
```
Comment (SAGARACH65) [inline admin-app/src/utils/httpHelper.ts]: Should we define a custom error for this?

**Pull request merged**